### ISSUE TITLE
Create a preview for HEIC (and possibly other types of images) for faster rendering

### ISSUE THREAD

**Issue by fsicoli** [sepinf-inc/IPED#262]
I suggest creating a feature that generates and stores previews of large image files. 

For instance, PSD files greater than 10 MB would be processed by imagemagick to create a lower resolution JPEG preview, stored directly on "indexador\view\" or in a SQLlite database.

That way, those large image files would be displayed faster in the analyst's workstations thus requiring less computer resources. The original high resolution file would still be able to be opened externally by double-clicking it.

**Comment by tc-wleite** [comment]
That seems a nice feature! 
I considered a similar idea myself a while ago, but didn't go further. 
As far as I know, for some formats (like JPEG), subsampling is used when reading large files, so the displayed image is already limited. However, even for these formats, very large images are somewhat slow to read.
Large PSDs and other formats (like HEIC) are particulary slow to open in the viewer, so they would be good candidates for having a preview image.

**Comment by lfcnassif** [comment]
Just to note #174 aims to get rid of thousands of small preview files in "view" folder and store them in sqlite containers like image thumbs currently are. So I think this could be implemented using the new container approach, otherwise will increase the effort needed for #174

**Comment by patrickdalla** [comment]
Hi, I've noted that HEIC images are not processed by FaceRecognitionTask(right @hauck-jvsh ?). Made this JPEG conversion can solve this also.

**Comment by wladimirleite** [comment]
Hi @lfcnassif!
Together with the "complaint" mentioned in https://github.com/sepinf-inc/IPED/issues/2414#issuecomment-2641093432, some users here (including myself) are not very happy with the time taken to render HEIC images.
There are some cases (usually iOS extractions) that contain several relevant images and when one of them is selected, it takes a few seconds (usually 2 to 5 in my PC, with the case on a very fast storage) to render. Just for comparison, large JPEG images (with a similar dimensions and file size), usually take 1 second or less.

My idea would be creating a JPG "view" of images of some mime types (configurable, initially only for HEICs) during processing.
Part of the processing cost is already paid when ImageThumbTask deals with such files, so the only drawback would be increasing the space used by the case. This should be small, as although HEIC images are somewhat common, usually there are not many of them.
What are your thoughts about this?

About the implementation, it could be just a new option in ImageThumbTask or a new task that would run before ImageThumbTask (so it doesn't need to decode the HEIC image again).

**Comment by lfcnassif** [comment]
I agree, HEIC images are taking some seconds to render, and when viewing thousands of them, that can be very annoying. Making mimetypes configurable would be nice! And maybe this can help running face recognition on HEIC, as @patrickdalla suggested.

**Comment by wladimirleite** [comment]
I am changing the issue description a bit, to handle configurable mime types.
We can add a minimum file size later, but even the smaller HEIC's I found (in terms of file size) take some time to render (because ImageMagick takes a while to decode them).

**Comment by lfcnassif** [comment]
> [@lfcnassif](https://github.com/lfcnassif), what about the implementation, what do you prefer?
> An option inside ImageThumbTask or a new task?

Not sure, I think a new option in current Task. It could be seen as a huge thumb. But if it's possible to reuse code, I'm fine with the other approach too.
